### PR TITLE
Feature/project set up

### PR DESCRIPTION
**Pull request by Rup-Narayan-Rajbanshi**
## Changes
  - Set up project

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+FROM python:3.12-slim-bullseye as base
+
+LABEL maintainer="Togglecorp Dev"
+
+ENV PYTHONUNBUFFERED 1
```
Comment (thenav56) [inline Dockerfile]:
```suggestion
ENV PYTHONUNBUFFERED=1
```

File: main/settings.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+from pathlib import Path
+
+import environ
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+env = environ.Env(
+    DJANGO_SECRET_KEY=str,
+    DJANGO_DEBUG=(bool, True),
```
Comment (thenav56) [inline main/settings.py]:
```suggestion
    DJANGO_DEBUG=(bool, False),
```

File: main/settings.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+WSGI_APPLICATION = "main.wsgi.application"
+
+
+# Database
+# https://docs.djangoproject.com/en/5.1/ref/settings/#databases
+
+DATABASES = {
+    "default": {
+        "ENGINE": "django.db.backends.postgresql",
```
Comment (thenav56) [inline main/settings.py]: We will need to install postgis.
https://docs.djangoproject.com/en/5.1/ref/contrib/gis/tutorial/#configure-settings-py

File: scripts/run_develop.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+python manage.py migrate &
```
Comment (thenav56) [inline scripts/run_develop.sh]: Let's not add migrate here, let's add it to readme instead.

As migrate command can be destructive, it is a good idea to do this manually :D

**Review by thenav56 — CHANGES_REQUESTED** [review]
Looks good. Few minor changes

File: main/settings.py (hunk 3 above)
Comment (Rup-Narayan-Rajbanshi) [reply main/settings.py]: @thenav56 I think we need to update in docker-compose.yml for db image as `postgis/postgis:15-3.5⁠` as well.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        # Helper packages
+        wait-for-it \
+    # Upgrade pip and install python packages for code
+    && pip install --upgrade --no-cache-dir pip poetry \
+    && poetry --version \
+    # Configure to use system instead of virtualenvs
+    && poetry config virtualenvs.create false \
+    && poetry install --no-root \
+    # Clean-up
+    && pip uninstall -y poetry virtualenv-clone virtualenv \
```
Comment (susilnem) [inline Dockerfile:25]: Let's add this one too
```suggestion
    && rm -rf /root/.cache/pypoetry \
    && pip uninstall -y poetry virtualenv-clone virtualenv \
```

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,30 @@
+FROM python:3.12-slim-bullseye as base
+
+LABEL maintainer="Togglecorp Dev"
+
+ENV PYTHONUNBUFFERED=1
```
Comment (susilnem) [inline Dockerfile:5]:
```suggestion
ENV PYTHONUNBUFFERED 1
```

**Review by susilnem — CHANGES_REQUESTED** [review]
Looks good.
Let's add a separate `.python-version` file for the python version which is `3.12`

**Review by thenav56 — APPROVED** [review]
LGTM

**Pull request merged**